stdlib/public/Synchronization/Mutex/LinuxImpl.swift

```diff
@@ -235,8 +235,7 @@ extension _MutexHandle {
 
     // EDEADLK - "The futex word at uaddr is already locked by the caller."
     case 35:
-      // TODO: Replace with a colder function / one that takes a StaticString
-      fatalError("Attempt to try to lock Mutex in already acquired thread")
+      return false
```

Comment:
was this just due to copy/past from the `lockSlow()` implementation? also, given that everything else just hits the default case, should this also just go there?

Reply:
I'd have to defer to @Azoy there. I wanted to keep the diff simple/readable but yes, this case ends up just being equivalent to `default` below.